Core/HLE/sceKernelModule.cpp

```diff
@@ -263,11 +269,23 @@ class PSPModule : public KernelObject {
 
 	void DoState(PointerWrap &p) override
 	{
-		auto s = p.Section("Module", 1, 4);
+		auto s = p.Section("Module", 1, 5);
 		if (!s)
 			return;
 
-		Do(p, nm);
+		if (s >= 5) {
+			Do(p, nm);
+		} else {
+			char temp[192];
+			NativeModule *pnm = &nm;
+			char *ptemp = temp;
+			DoArray(p, ptemp, 0xC0);
+			memcpy(pnm, ptemp, 0x2C);
+			pnm->modid = GetUID();
+			pnm += 0x30;
+			ptemp += 0x30;
+			memcpy(pnm, ptemp, 0xC0 - 0x2C);
+		}
 		Do(p, memoryBlockAddr);
```

Comment:
why not just put 'modid' last in 'NativeModule' to keep the struct backwards compatible and avoid this hack ? or is 'NativeModule'  something the real PSP OS uses and it just had wrong offsets before ?

Reply:
NativeModule is a native struct of PSP, its members has fixed order.

Reply:
ok I see.
why are you increasing both `pnm`and `ptemp` by `0x30` though ? shouldn't `ptemp`be increased by `0x2C` instead ?

Reply:
Oops, thanks.

Reply:
Turns out the math was wrong - have to cast the pointers to bytes first :)

Fixed in 2f4dfed

Reply:
Oops again, thanks:p